src/VisualStudio/Core/Def/Implementation/SymbolSearch/RoslynSymbolResult.cs

```diff
@@ -0,0 +1,213 @@
+
+            result.ClassifiedDefinition = new ClassifiedTextElement(
+                definition.NameDisplayParts.Select(n =>
+                    new ClassifiedTextRun(n.Tag.ToClassificationTypeName(), n.ToVisibleDisplayString(includeLeftToRightMarker: true))));
+            /*
+            result.ClassifiedContext = new ClassifiedTextElement(
+                definition.DisplayParts.Select(n =>
+                    new ClassifiedTextRun(n.Tag.ToClassificationTypeName(), n.ToVisibleDisplayString(includeLeftToRightMarker: true))));
+            */
+
+            var (projectId, projectName, sourceText, documentId) = await GetGuidAndProjectNameAndSourceTextAsync(documentSpan.Document)
+                .ConfigureAwait(false);
+            result.ProjectId = projectId.ToString();
+            result.ContainerName = projectName;
+            result.DocumentId = documentId.ToString();
+
+            var (excerptResult, lineText) = await ExcerptAsync(sourceText, documentSpan, token).ConfigureAwait(false);
+
+            // decorate the parent definition:
+            result.ClassifiedDefinition = await ProtocolConversions.DocumentSpanToLocationWithTextAsync(definition.SourceSpans.First(), text, cancellationToken).ConfigureAwait(false);
+            result.DefinitionIcon = definition.Tags.GetFirstGlyph().GetImageId();
+
+            // do this for each reference:
+            var classifiedSpansAndHighlightSpan = await ClassifiedSpansAndHighlightSpanFactory.ClassifyAsync(reference.SourceSpan, context.CancellationToken).ConfigureAwait(false);
+            var classifiedSpans = classifiedSpansAndHighlightSpan.ClassifiedSpans;
+            var referenceLocation = await ProtocolConversions.DocumentSpanToLocationAsync(reference.SourceSpan, cancellationToken).ConfigureAwait(false);
+            var docText = await reference.SourceSpan.Document.GetTextAsync(context.CancellationToken).ConfigureAwait(false);
+            var classifiedText = new ClassifiedTextElement(classifiedSpans.Select(cspan => new ClassifiedTextRun(cspan.ClassificationType, docText.ToString(cspan.TextSpan))));
+            var locationWithText = new LSP.LocationWithText { Range = referenceLocation.Range, Uri = referenceLocation.Uri, Text = classifiedText };
+
+            result.ClassifiedContext = 
+            var mappedDocumentSpan = await TryMapAndGetFirstAsync(documentSpan, sourceText, token).ConfigureAwait(false);
+            if (mappedDocumentSpan.HasValue)
+            {
+                var location = mappedDocumentSpan.Value;
+                result.PersistentSpan = symbolSource.ServiceProvider.PersistentSpanFactory.Create(
+                    documentSpan.Document.FilePath,
+                    location.LinePositionSpan.Start.Line,
+                    location.LinePositionSpan.Start.Character,
+                    location.LinePositionSpan.End.Line,
+                    location.LinePositionSpan.End.Character,
+                    SpanTrackingMode.EdgeInclusive);
```

Comment:
what does it mean to map locations to an arbitrary file path? how could that possibly work in a multi-threaded concurrent world?